client/src/utilities/chart.js

```diff
@@ -1,6 +1,11 @@
+import Immutable from 'immutable';
+
 export function getChartData(visualisation, datasets) {
   const { datasetId, spec } = visualisation;
-  const dataset = datasets[datasetId];
```

Comment:
It's currently not the case when viewing a shared visualisation or dashboard - as far as I can see these functions are both broken in develop as a result